salt/_states/metalk8s_kubernetes.py

```diff
@@ -472,6 +472,58 @@ def namespace_present(name, **kwargs):
     return ret
 
 
+def namespace_annotation_present(
+        name,
+        annotation_key,
+        annotation_value,
+        **kwargs):
+    '''
+    Ensures that the namespace is patched.
+
+    name
+        The name of the namespace
+
+    annotation_key
+        Key of the annotation
+    annotation_value
+        Value of the annotation
+    '''
+
+    ret = {'name': name,
+           'changes': {},
+           'result': False,
+           'comment': ''}
+
+    annotation = __salt__['metalk8s_kubernetes.namespace_add_annotation'](
+        name, annotation_key, annotation_value, **kwargs
+    )
```

Comment:
You are not supporting test mode here.
Also it's not idempotent.

You need to add a module to list the annotation of a specific namespace, then you check if you annotation already there with the right value if not you update/create it.
Check all the the other state for example `node_annotation_present` the code should be really close only the function you call are different.

Reply:
ok